data/lua/core/wml.lua

```diff
@@ -345,6 +366,7 @@ if wesnoth.kernel_type() ~= "Application Lua Kernel" then
 
 	local root_variable_mt = {
 		__metatable = "WML variables proxy",
+		__dir = dir_vars(true),
```

Comment:
I think it would be the other way around – the tab completion could be made to use the dir mechanism instead of what it currently does. That doesn't mean calling the dir() function but rather using the underlying implementation that dir() uses. It also means tab-completion will not reveal deprecated functions, but that might be fine?

Reply:
Have you already looked at the existing mechanism, and if so please would you give an overview of what existing functions are gaining parallel implementations in this PR?

(I'll have a look for myself, but the review can be quicker if I you give me a quick-start on that.)